### PR TITLE
Add new UI for batcher holding

### PR DESCRIPTION
**Pull request by kienle371999**
In this PR, I added the UI of batcher holding to compute the tokens based on the `pro-rata model` in the smart contract.

### CODE REVIEW

**Comment by kienle371999** [comment]
I will rename the `NewHoldings module` to `Holdings module` and delete the old `Holdings module` after I finish everything @glottologist.

File: batcher-ui/src/pages/Main.tsx [hunk 1]
```diff
@@ -82,23 +84,98 @@ const Welcome: React.FC = () => {
...
+        bigMapsByIdUri + storage.batch_set.batches + '/keys/' + openOrderBooks.at(i).batch_number;
+      const batchData = await fetch(batchURI, { method: 'GET' });
+      const batch = await batchData.json();
+      console.log(1444, batch);
+
+      const clearingRate =
+        parseInt(batch.value.status.cleared.rate.rate.val) *
+        10 ** parseInt(batch.value.status.cleared.rate.rate.pow);
+
+      const clearingKey = Object.keys(batch.value.status.cleared.clearing.clearing_tolerance)[0];
```
Comment (glottologist) [inline batcher-ui/src/pages/Main.tsx:132]: We need to check if their order was in those levels that were interested in the clearing price.  If they are outside of the clearing price then they don't get filled and we can just transfer their original amount without doing any further calculation.  Logic for this is in the contract treasury

File: batcher-ui/src/pages/Main.tsx [hunk 2]
```diff
@@ -82,23 +84,98 @@ const Welcome: React.FC = () => {
...
+      const buySideActualVolume = parseInt(
+        batch.value.status.cleared.clearing.prorata_equivalence.buy_side_actual_volume,
+      );
+      const sellSideActualVolume = parseInt(
+        batch.value.status.cleared.clearing.prorata_equivalence.sell_side_actual_volume,
+      );
+
+      if (Object.keys(openOrderBooks.at(i).side)[0] === 'bUY') {
+        const depositedBuySideAmount = parseInt(openOrderBooks.at(i).swap.from.amount);
+        const unconvertedBuySideAmount =
```
Comment (glottologist) [inline batcher-ui/src/pages/Main.tsx:148]: Not sure this is quite correct. 
1. We want to calculate how much (% or ratio) this order was in terms of all orders placed.  I.e this order was 10% of the whole buy volume placed in that batch.
2. We then know how much of the CLEARED volume that person should get.  I.e. if the cleared volume on the sell side was 10 mil USDT then they should get 10% of 10 mil USDT. 
3. We then need to work out how much of the buy side they have remaining.   (10% of 10 mil USDT) converted to tzBTC with the clearing price should always be LESS or EQUAL TO their original order amount in tzBTC.  
4. If LESS then it is a partial fill and then get their 10% of 10 mil USDT PLUS whatever they have remaining on the tzBTC side afer we have subtracted the equivalent in tzBTC from their order deposit amount. 
5. If EQUAL, then it is a COMPLETE fill and they just get their 10% of 10 mil USDT.   This however is less likely than a partial fill.

**Review by glottologist — CHANGES_REQUESTED** [review]
please see comments

File: batcher-ui/src/pages/Main.tsx (hunk 1 above)
Comment (kienle371999) [reply batcher-ui/src/pages/Main.tsx:128]: I get the `batches` that this current user includes in the open section and get the exciting rate of this batch.

File: batcher-ui/src/pages/Main.tsx (hunk 2 above)
Comment (kienle371999) [reply batcher-ui/src/pages/Main.tsx:144]: Yup. I computed the exchanged tokens and remaining tokens based on the ratio. You can see it again in my new commits.

**Comment by kienle371999** [comment]
I added the codebase to represent the correct holder tokens of each wallet. You can see it, @glottologist. However, I wanna change the re-rendering in the holdings to improve the user experience.

**Comment by kienle371999** [comment]
> Can you change this before I review @kienle371999

I removed the `NewHoldings`, @glottologist

**Review by glottologist — APPROVED** [review]
Let's get this merged and deployed to Kathmandunet

**Pull request merged**